File: quote/Projects/PluginModel/bom/NoteProperties.vb
Imports System.ComponentModel

Namespace BOM

    Public Class NoteProperties
        Inherits Common.NoteProperties

        Private _QuoteHeader As Header
        Private _Note As String
        Private _Note2Customer As String

        Public Sub New(ByVal QuoteHeader As Header)
            _QuoteHeader = QuoteHeader
        End Sub
        <DisplayName("Note-Internal"), _
         CategoryAttribute("Notes")> _
        Public Property Note As String
            Get
                Return _Note
            End Get
            Set(ByVal value As String)
                If _Note <> value Then
                    _Note = value
                    Me.SendEvents()
                End If
            End Set
        End Property
        <DisplayName("Note-To Customer"), _
         CategoryAttribute("Notes")> _
        Public Property Note2Customer As String
            Get
                Return _Note2Customer
            End Get
            Set(ByVal value As String)
                If _Note2Customer <> value Then
                    _Note2Customer = value
                    Me.SendEvents()
                End If
            End Set
        End Property

    End Class
End Namespace

File: quote/Projects/PluginModel/template/NoteProperties.cs
namespace Model.Template
{
    using System;
    using System.ComponentModel;
    using System.Linq;

    /// <summary>
    /// A class that contains properties for the notes 
    /// of the quote.
    /// </summary>
    public sealed class NoteProperties : Common.NoteProperties
    {
        /// <summary>
        /// The note string.
        /// </summary>
        private string note;

        /// <summary>
        /// The note for the customer.
        /// </summary>
        private string note2Customer;

        /// <summary>
        /// Initializes a new instance of the <see cref="NoteProperties" /> class.
        /// </summary>
        public NoteProperties()
        {
        }

        /// <summary>
        /// Gets or sets the text for the note.
        /// </summary>
        /// <value>The text of the internal note.</value>
        [DisplayName("Note-Internal"), CategoryAttribute("Notes")]
        public string Note 
        {
            get 
            { 
                return this.note; 
            }
            
            set 
            {
                if (this.note != value) 
                {
                    this.note = value;
                    this.OnPropertyChanged();
                }
            }
        }

        /// <summary>
        /// Gets or sets the note to the customer.
        /// </summary>
        /// <value>The text of the note to the customer.</value>
        [DisplayName("Note-To Customer"), CategoryAttribute("Notes")]
        public string Note2Customer 
        {
            get 
            { 
                return this.note2Customer; 
            }
            
            set 
            {
                if (this.note2Customer != value) 
                {
                    this.note2Customer = value;
                    this.OnPropertyChanged();
                }
            }
        }
    }
}
